### ISSUE TITLE
Slope sprite drawing problem?

### ISSUE THREAD

**Issue by Meerschweinmann** [team-eternity/eternity#637]
Hello again :)

Because i know that the slope code is complete and in a testing phase, this older problem i had posted at Doomworld came back in my mind, and i thought that could be a slope problem with sprite drawing.
The sprites can be seen through the sloped ceiling.

This behavior can be reproduced when loading my example map with the latest DRD devbuild, using the KILLEM cheat and walk into the house below the slope.

Here is the testmap:

[line-portaltest.zip](https://github.com/user-attachments/files/17102806/line-portaltest.zip)

And here a picture showing whats wrong:

![etrn12](https://github.com/user-attachments/assets/f7966d4d-933d-479c-8c77-1e8f4f78748f)

**Comment by Meerschweinmann** [comment]
Update. Only the dead enemies can be seen through the sloped ceiling. When alive the sprites are handled as they should

**Comment by ioan-chera** [comment]
Yes, this is the natural consequence of classic DOOM allowing sprites to fully draw when their things are not mechanically occluded. This is needed when you have multiple adjacent sectors with the same floor height; you don't want actors' sprites to be clipped by every separation when there's no change in height, unless they're actually physically below.

What you get here is related, except that the slope skews all assumptions, especially because now it works like a wall. I can make them block absolutely all sprites, regardless of mechanical range, but this might result in the unwanted artifacts mentioned above. Clearly this needs more care.

**Comment by Meerschweinmann** [comment]
Hi Ioan,

sounds like a not so easy task to solve this behavior, because it is something it was originally intended to behave so back in the 90s.

And meanwhile i tested around a bit with the slope and confirm that it has to do something with the sprite/slope height.

When lowering the slopes lowest point 4 pixels more, the dead sprites are shown as they should and only the ammo clips are showing through the slope.

![etrn22](https://github.com/user-attachments/assets/05b7e756-788f-47ee-95d7-c1aa56b26aef)

**Comment by ioan-chera** [comment]
Hopefully fixed, let's hope no ill side effects.

**Comment by Meerschweinmann** [comment]
Thank you. 

Will have a look tomorrow when the new DRD devbuild is downloadable.
Don't know why, but this time compiling with MSYS showed some errors and aborted.

**Comment by Meerschweinmann** [comment]
Sprite drawing while standing below the slope works great now 👍
